### PR TITLE
feat: Use free_range to free memory

### PR DESCRIPTION
**Pull request by holykol**
also bump to 0.2.0
also rename package to `gear-dlmalloc`

@breathx 
@grishasobol 
@shamilsan

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
-name = "dlmalloc"
-version = "0.1.4"
+name = "gear-dlmalloc"
+version = "0.2.0"
 authors = ["Alex Crichton <alex@alexcrichton.com>"]
```
Comment (shamilsan) [inline Cargo.toml:4]: Need to add Gear 😄

**Review by shamilsan — APPROVED** [review]
Also need to rename the repo and write a couple of words in `README.md`

**Review by grishasobol — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (holykol) [reply Cargo.toml]: Thanks, added. Also specified this is a fork in the readme

**Review by shamilsan — APPROVED** [review]
Awesome 🔥

**Pull request merged**